File: LeetCode_Solutions/LeetCode_Solutions/HammingDistance_461.m
//
//  HammingDistance_461.m
//  LeetCode_Solutions
//
//  Created by Huni on 17/03/2017.
//  Copyright © 2017 KnightJoker. All rights reserved.
//

#import "HammingDistance_461.h"



@implementation HammingDistance_461

- (NSInteger)hammingDistance:(NSInteger)x
                         and:(NSInteger)y {
    
    NSInteger res = 0;
    NSInteger exc = x ^ y;
    while (exc) {
        ++res;
        exc &= (exc - 1);
    }
    return res;
    
}

@end

File: LeetCode_Swift/LeetCode_Swift/HammingDistance_461.swift
//
//  HammingDistance_461.swift
//  LeetCode_Swift
//
//  Created by Huni on 24/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class HammingDistance_461 {

    func hammingDistance(_ x: Int, _ y: Int) -> Int {
        
        var res = 0
        var exc = x ^ y
        while exc != 0 {
            res += 1
            exc &= (exc - 1)
        }
        return res
    }
}
